src/main/java/cz/cvut/kbss/termit/service/business/SearchService.java

```diff
@@ -38,9 +43,18 @@ public SearchService(SearchDao searchDao) {
      * Executes full text search in assets.
      *
      * @param searchString String to search by
+     * @param vocabularies URIs of vocabularies to search in, or null, if all vocabularies shall be searched
+     * @param termsOnly whether to return only matches related to terms (not vocabularies)
      * @return Matching assets
      */
-    public List<FullTextSearchResult> fullTextSearch(String searchString) {
-        return searchDao.fullTextSearch(searchString);
+    public List<FullTextSearchResult> fullTextSearch(String searchString, Set<URI> vocabularies, boolean termsOnly) {
+        Stream<FullTextSearchResult> result = searchDao.fullTextSearch(searchString).stream();
+        if (termsOnly) {
+            result = result.filter(r -> r.getTypes().contains(SKOS.CONCEPT));
+        }
+        if (vocabularies != null) {
+            result = result.filter(r -> vocabularies.contains(r.getVocabulary()));
```

Comment:
Possible (but admittedly unlikely) NPX in case termsOnly is false, vocabularies is not null and r is a vocabulary. In that case, contains would be called with null, which for some collection may result in NPX.